includes/Checker/Checks/Localhost_Check.php

```diff
@@ -43,16 +43,31 @@ public function get_categories() {
 	 */
 	protected function check_files( Check_Result $result, array $files ) {
 		$php_files = self::filter_files_by_extension( $files, 'php' );
-		$file      = self::file_preg_match( '#https?://(localhost|127.0.0.1)#', $php_files );
-		if ( $file ) {
-			$result->add_message(
-				true,
-				__( 'Do not use Localhost/127.0.0.1 in your code.', 'plugin-check' ),
-				array(
-					'code' => 'localhost_code_detected',
-					'file' => $file,
-				)
-			);
+
+		foreach ( $php_files as $file ) {
+			$code   = file_get_contents( $file );
+			$tokens = token_get_all( $code );
```

Comment:
Such dedicated PHPCS sniff does not exist. I suggested for such sniff here https://github.com/PHPCSStandards/PHPCSExtra/issues/302 but it has been closed as this would too generalized and very difficult to maintain and support. So I believe PHPCS approach is no go this scenario.